crossdock/client/behavior/run.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package behavior
+
+// Run the given function inside a behavior context.
+//
+// Functions like Fatalf won't work if the behavior is not executed inside a
+// Run context.
+func Run(f func()) {
+	done := make(chan struct{})
+
+	// We run the function inside a goroutine so that Fatalf can simply call
+	// runtime.Goexit to stop execution.
+	go func() {
+		defer func() {
```

Comment:
can avoid inline function for the channel write by using `close` instead. i actually prefer close for cases like this:
 - releases resources for the channel immediately
 - panics if there's a code bug which causes it to try to close twice

```go
go func() {
  defer close(done)
  f()
}()
```